aea/skills/base.py

```diff
@@ -877,15 +896,16 @@ def _filter_classes(
         """
         Filter classes of skill components.
 
+        The following filters are applied:
+        - the class must be a subclass of "SkillComponent";
+        - its __module__ attribute must not start with 'aea.' (we exclude classes provided by the framework)
+        - its __module__ attribute starts with the expected dotted path of this skill.
+
         :param classes: a list of pairs (class name, class object)
         :return: a list of the same kind, but filtered with only skill component classes.
         """
         filtered_classes = filter(
-            lambda name_and_class: any(
-                re.match(component, name_and_class[0])
-                for component in self._all_component_names
-            )
-            and issubclass(name_and_class[1], SkillComponent)
```

Comment:
if we apply this filter, we wouldn't get any class that is not declared, and so no warning message